### PR TITLE
Fix Cydia/iOS packaging and compilation issues ##build

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/flag/Makefile [hunk 1]
```diff
@@ -4,4 +4,5 @@ NAME=r_flag
 R2DEPS=r_util
 OBJS=flag.o zones.o tags.o
 
+include $(SHLR)/zip/deps.mk
```
Comment (ret2libc) [inline libr/flag/Makefile:7]: why this? on meson r_flag depends only on r_util (which in turn depends on zip)

File: binr/blob/Makefile [hunk 2]
```diff
@@ -90,7 +91,7 @@ build:
 		rm -f _$${FILE} ; \
 	done
 	rm -f _*
-	${CC} -o radare2 ${SOURCES} ${CFLAGS} ${LDFLAGS}
+	${CC} -I$(SHLR)/sdb/src -o radare2 ${SOURCES} ${CFLAGS} ${LDFLAGS}
```
Comment (ret2libc) [inline binr/blob/Makefile:94]: this should not be necessary I think, because it should already be in CFLAGS after you include sdb.mk

**Comment by trufae** [comment]
Now it seems to fix sys/ios-cydia.sh

File: binr/rules.mk [hunk 3]
```diff
@@ -1,6 +1,7 @@
 BINR_PROGRAM=1
 include ../../libr/config.mk
 include ../../shlr/zip/deps.mk
+include ../../shlr/sdb.mk
```
Comment (ret2libc) [inline binr/rules.mk:4]: There is already this line. Why did you dupped it?

**Comment by XVilka** [comment]
Should be also cherry-picked to 4.5.1

File: libr/flag/Makefile (hunk 1 above)
Comment (trufae) [reply libr/flag/Makefile]: because the build was failing without this dependency

**Comment by XVilka** [comment]
Fails on FreeBSD (https://builds.sr.ht/~xvilka/job/263283#task-build-148):
```
gmake[4]: Entering directory '/usr/home/build/radare2/shlr/sdb'
echo '#define SDB_VERSION "1.4.1"' > src/sdb_version.h
gmake[4]: Leaving directory '/usr/home/build/radare2/shlr/sdb'
gcc -c  -fPIC  -o cdb.o cdb.c
gmake[3]: gcc: No such file or directory
gmake[3]: *** [Makefile:75: cdb.o] Error 127
gmake[3]: Leaving directory '/usr/home/build/radare2/shlr/sdb/src'
gmake[2]: *** [Makefile:154: sdb-native-build] Error 2
gmake[2]: Leaving directory '/usr/home/build/radare2/shlr'
gmake[1]: *** [Makefile:131: targets] Error 2
gmake[1]: Leaving directory '/usr/home/build/radare2/shlr'
gmake: *** [Makefile:56: all] Error 2
```
It hardcodes `gcc` somewhere.

**Comment by trufae** [comment]
iirc gcc is only hardcoded in sys/ios-cydia.sh which it should run only on mac not bsd

**Comment by XVilka** [comment]
Error is still here, seems SDB assumes for some reason GCC as a compiler after these changes:
```
gmake[4]: Entering directory '/usr/home/build/radare2/shlr/sdb'
echo '#define SDB_VERSION "1.4.1"' > src/sdb_version.h
gmake[4]: Leaving directory '/usr/home/build/radare2/shlr/sdb'
gcc -c  -fPIC  -o cdb.o cdb.c
gmake[3]: gcc: No such file or directory
gmake[3]: *** [Makefile:75: cdb.o] Error 127
gmake[3]: Leaving directory '/usr/home/build/radare2/shlr/sdb/src'
gmake[2]: *** [Makefile:154: sdb-native-build] Error 2
gmake[2]: Leaving directory '/usr/home/build/radare2/shlr'
gmake[1]: *** [Makefile:131: targets] Error 2
gmake[1]: Leaving directory '/usr/home/build/radare2/shlr'
gmake: *** [Makefile:56: all] Error 2
```

**Comment by trufae** [comment]
I think this hardcoded gcc issue is fixed in https://github.com/radareorg/radare2/pull/17141

i can apply the fix in both PRs and rebase when merged to solve the conflicts or just wait for the 17141 to be merged

**Comment by trufae** [comment]
I think this PR is now good to merge

**Comment by XVilka** [comment]
Just cherry-pick that change here, that LTO is tricky beast and it's unknown when will be fixed. This PR is more important.

**Comment by XVilka** [comment]
Cherry-picked into [4.5.1](https://github.com/radareorg/radare2/milestone/57).

**Pull request merged**